### ISSUE TITLE
Bug: Eager svd/eigh hardcode eps=0.0, producing NaN gradients (traced uses 1e-12)

### ISSUE THREAD

**Issue by terasakisatoshi** [tensor4all/tenferro-rs#1232]
## Summary

Eager `svd` and `eigh` build their linalg ops with `eps: 0.0`, which disables the
singularity/degeneracy regularization that the AD rules rely on. As a result,
reverse-mode gradients of the eager SVD / eigh **vectors** are `NaN`, whereas the
traced API (which uses `eps = 1e-12`) returns finite gradients.

## Affected crate / location

- `crates/tenferro-linalg/src/eager_ext.rs:124` — `LinalgOp::Svd { eps: 0.0 }`
- `crates/tenferro-linalg/src/eager_ext.rs:361` — `LinalgOp::Eigh { eps: 0.0 }`
- Contrast: `crates/tenferro-linalg/src/traced.rs:160` (`svd` → `svd_with_eps(a, 1e-12)`)
  and `crates/tenferro-linalg/src/traced.rs:222` (`eigh` → `eigh_with_eps(a, 1e-12)`)
- AD rule using `eps`: `crates/tenferro-linalg/src/ad/rules/mod.rs:461-465` (SVD),
  and the analogous `eigh` block (`diff` / `diff_sq` / `eps_sq` / `safe_diff` / `f`).

## Current behavior

```rust
// eager_ext.rs
pub fn svd(a: &EagerTensor) -> Result<...> {
    let mut outputs = apply_linalg_eager(LinalgOp::Svd { eps: 0.0 }, &[a])?.into_iter();
    ...
}
// eigh similarly uses LinalgOp::Eigh { eps: 0.0 }
```

```rust
// ad/rules/mod.rs (SVD)
let s_gap   = fixed_mul(...);                 // diagonal entry == 0
let s_gap_sq = fixed_mul(s_gap, s_gap);       // == 0
let eps_sq  = fixed_scale(ones_mat, eps * eps); // == 0 when eps == 0.0
let safe_gap = fixed_add(s_gap_sq, eps_sq);   // == 0
let f       = fixed_div(s_gap, safe_gap);     // 0 / 0 == NaN
```

## Why this is a bug

`eps` is unused by the forward LAPACK call (so eager forward values are correct),
but it is the guard term in the AD rules. On the diagonal of the gap matrix the
singular-value gap is exactly `0`, so `f_diag = 0 / (0 + eps^2)`. With the intended
`eps = 1e-12` this is `0` (finite); with `eps = 0.0` it is `0/0 = NaN`. The `NaN`
then propagates through `hadamard(f, ...)` into the `U`/`V` (SVD) and eigenvector
(`eigh`) cotangents for *every* input, not just degenerate ones. The eager API
never exposes an `eps`, so callers cannot work around it.

## Reproduction / trigger

Build an `EagerTensor` with `requires_grad`, call `.svd()` (or `.eigh()`), form a
loss that depends on `U`/`V` (or eigenvectors), call `.backward()`, and inspect the
gradient — it contains `NaN`. The same computation through the traced API
(`eps = 1e-12`) yields finite gradients. (The existing eager test only asserts
`grad().is_some()`, so it does not catch the `NaN`.)

## Suggested fix

Use the same default epsilon as the traced API (`eps: 1e-12`) in `eager_ext.rs`
for `Svd` and `Eigh`, and consider exposing `*_with_eps` eager variants for parity.

## Severity / confidence

Severity: high (silent NaN gradients on a primary AD surface). Confidence: high.

---
> 🔍 Found via source analysis performed with **Claude Opus 4.8 (1M context)**.
